### PR TITLE
Adding eslint

### PR DESCRIPTION
**Pull request by Rohitbels**
- Added linting support using Eslint.
- Extended configs eslint-config-standard, eslint-config-prettier
- Fixed all linting issues found.
- Skipped websites, examples  and docs-e2e-testing directory.

## Summary/ Motivation (TLDR;)
Making sure better and more readable code.

## Related issues
Add eslint for development
https://github.com/nvh95/jest-preview/issues/103

### CODE REVIEW

**Comment by Rohitbels** [comment]
@nvh95

File: cli/configCra.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ try {
     2,
   )}\n`;
   fs.writeFileSync('jest.config.js', jestConfigFileContent);
-  console.log(`Added jest.config.js to the project.`);
+  console.log('Added jest.config.js to the project.');
```
Comment (nvh95) [inline cli/configCra.js:43]: Can you tell me which rule of eslint is this?

File: cli/clearCache.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ function clearJestCache() {
   }
 }
 
-function clearJestPreviewCache() {
+function clearJestPreviewCache () {
```
Comment (nvh95) [inline cli/clearCache.js:24]: Can you tell me which rule of eslint is this?

File: cli/configCra.js [hunk 3]
```diff
@@ -69,11 +69,11 @@ try {
       )
       // Require `env` direct from `react-scripts`
       .replace(
-        `require('../config/env');`,
-        `require('react-scripts/config/env');`,
+        "require('../config/env');",
+        "require('react-scripts/config/env');",
```
Comment (nvh95) [inline cli/configCra.js:73]: Can you tell me which rule of eslint is this?

File: jest.config.js [hunk 4]
```diff
@@ -20,7 +20,6 @@ module.exports = {
   transformIgnorePatterns: [
     '[/\\\\]node_modules[/\\\\].+\\.(js|jsx|mjs|cjs|ts|tsx)$',
   ],
-  modulePaths: ['<rootDir>/demo'],
```
Comment (nvh95) [inline jest.config.js:23]: Why do we remove this?

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        "node": true,
+        "jest": true
+    },
+    "extends": [
+        "plugin:react/recommended",
+        "plugin:react/jsx-runtime",
+        "standard",
+        "prettier"
+    ],
+    "ignorePatterns": ["website/**/*", "examples/**/*", "docs-e2e-testing/**/*"],
```
Comment (nvh95) [inline .eslintrc.json:14]: Can you update it to?
```suggestion
    "ignorePatterns": ["website/**/*", "examples/**/*", "docs-e2e-testing/**/*", "transforms/**/*", "dist/**/*", "playground/**/*"],
```

**Review by nvh95 — CHANGES_REQUESTED** [review]
Hi @Rohitbels. Thank you very much for spending time filing this PR. Below is some of my feedback.

- There are some updates I am not sure which `eslint` rule it refers to. Like 
```diff
-  console.log(`Added jest.config.js to the project.`);
+  console.log('Added jest.config.js to the project.');
```

and 
```diff
-function clearJestCache() {
+function clearJestCache () {
```

Can you specify which rule is it? I did try it on my computer but `eslint` does not show any warnings

- I am working on #255, which will restructure the whole repository. It would be easier if your PR contains only the `eslint` configuration, not the update made by `eslint`. We will run the `eslint` against the whole source code when #255 gets merge, that would reduce the conflicts we might encounter. Can you help to revert it back? Thanks.

File: cli/clearCache.js (hunk 2 above)
Comment (Rohitbels) [reply cli/clearCache.js]: Added linting support using eslint and fixing errors.

File: cli/configCra.js (hunk 3 above)
Comment (Rohitbels) [reply cli/configCra.js]: Revert all linting changes for now

File: .eslintrc.json (hunk 5 above)
Comment (Rohitbels) [reply .eslintrc.json]: Done

**Review by nvh95 — APPROVED** [review]
LGTM! Thank you again @Rohitbels

**Comment by nvh95** [comment]
@all-contributors please add @Rohitbels for code

**Comment by nvh95** [comment]
@Rohitbels Welcome to contributors 🎉

**Comment by Rohitbels** [comment]
Thank you @nvh95 this is an awesome module hope to contribute more.

**Pull request merged**